### PR TITLE
allow custom message text for email/slack notification

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

**Comment by robfletcher** [comment]
Example of notification JSON: 

```{
  "notifications": [
    {
      "level": "pipeline",
      "when": [
        "pipeline.failed"
      ],
      "type": "email",
      "address": "delivery-engineering@netflix.com",
      "message": {
        "pipeline.failed": {
          "text": "O NOES!!"
        }
      }
    }
  ]
}```

File: echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy [hunk 1]
```diff
@@ -16,90 +16,107 @@
...
-        sendMessage(
-            [preference.address] as String[],
-            event,
-            subject,
-            config.type,
-            status,
-            config.link
-        )
+  @Autowired
+  EmailNotificationService mailService
```
Comment (ajordens) [inline echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy]: PR party foul ... changing spaces

File: echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy [hunk 2]
```diff
@@ -16,90 +16,107 @@
...
+        executionId: event.content?.execution?.id,
+        type       : type,
+        status     : status,
+        link       : link,
+        name       : event.content?.execution?.name ?: event.content?.execution?.description
+      ]
+    )
+
+    if (customMessage) {
+      body += "\n\n" + customMessage
```
Comment (tomaslin) [inline echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy]: feels we want a better separator for this?

**Comment by tomaslin** [comment]
why sms get no love?

File: echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy (hunk 2 above)
Comment (ajordens) [inline echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy]: Emails are full on plain text right now aren't they?

File: echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy [hunk 3]
```diff
@@ -16,90 +16,107 @@
...
     }
```
Comment (robfletcher) [inline echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy]: append custom message to body from velocity template

File: echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy (hunk 2 above)
Comment (tomaslin) [inline echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy]: well, if they go onto gmail it should be able to render fun characters

Comment (robfletcher) [inline echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy]: Could pass it to Velocity & then we can render it any way we like

**Comment by ajordens** [comment]
LGTM (haven't tested the templates but seems reasonable given everything is plain text afaik)

File: echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy (hunk 2 above)
Comment (robfletcher) [inline echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EmailNotificationAgent.groovy]: Updated PR

**Comment by robfletcher** [comment]
@tomaslin I didn't do SMS because we could blow the character limit. If that's not a concern it would be easy to add.

**Pull request merged**